web/pages/home/index.tsx

```diff
@@ -324,6 +347,86 @@ function SearchSection(props: {
   )
 }
 
+function FeaturedSection() {
+  const [pinned, setPinned] = useState<JSX.Element[]>([])
+  const [posts, setPosts] = useState<Post[]>([])
+  const [homeConfig, setHomeConfig] = useState<HomeConfig | undefined>(
+    undefined
+  )
+  useEffect(() => {
+    getHomeConfig().then(setHomeConfig)
+    getAllPosts().then(setPosts)
+  }, [])
+
+  useEffect(() => {
+    const pinnedItems = homeConfig?.pinnedItems
+
+    async function getPinned() {
+      if (pinnedItems == null) {
+        if (homeConfig != null) {
+          updateHomeConfig(homeConfig, { pinnedItems: [] })
+        }
+      } else {
+        const itemComponents = await Promise.all(
+          pinnedItems.map(async (element) => {
+            if (element.type === 'post') {
+              const post = await getPost(element.itemId)
+              if (post) {
+                return <PostCard post={post as Post} />
+              }
+            } else if (element.type === 'contract') {
+              const contract = await getContractFromId(element.itemId)
+              if (contract) {
+                return <ContractCard contract={contract as Contract} />
+              }
+            }
+          })
+        )
+        setPinned(
+          itemComponents.filter(
+            (element) => element != undefined
+          ) as JSX.Element[]
+        )
+      }
+    }
+    getPinned()
+  }, [homeConfig])
+
+  async function onSubmit(selectedItems: { itemId: string; type: string }[]) {
+    if (homeConfig == null) return
+    await updateHomeConfig(homeConfig, {
+      pinnedItems: [
+        ...(homeConfig?.pinnedItems ?? []),
+        ...(selectedItems as { itemId: string; type: 'contract' | 'post' }[]),
+      ],
+    })
+    getHomeConfig().then(setHomeConfig)
```

Comment:
This is ok, but if someone else edited pinnedItems, it would only fetch it after you submit.

You don't have to change it, but a better way to do this is to listen for home config above with a hook. Then you wouldn't need to fetch it again after setting home config.